src/modules/site-settings/edit-site-details.tsx

```diff
@@ -90,8 +92,8 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 		setCustomDomain( selectedSite.customDomain ?? null );
 		setCustomDomainError( '' );
 		setIsChangeWpError( '' );
-		setEnableHttps( selectedSite.enableHttps ?? false );
-	}, [ currentWpVersion, selectedSite ] );
+		setEnableHttps( selectedSite.enableHttps ?? isCertificateTrusted );
```

Comment:
This change makes it so the HTTPS checkbox is enabled by default when I open the `Edit site` form for a site with a custom domain without HTTPS.

Not sure if that's intentional, but it seems like unexpected behavior to me.